tools/platyPS/GenerateHelp.ps1

```diff
@@ -0,0 +1,133 @@
+if ((Get-Module -ListAvailable -Name platyPS).Count -eq 0)
+{
+    Write-Host "Installing platyPS"
+    Install-Module -Name platyPS -scope CurrentUser
+    Import-Module platyPS
+}
+
+# If we can't find a path for the given service, then it doesn't exist in AzureRM
+if (!(Test-Path $PathToRepo\src\ResourceManager\$Service\Commands.$Service))
+{
+    throw "Cannot find path to $PathToRepo\src\ResourceManager\$Service\Commands.$Service"
+}
+
+# Import service module
+Import-Module "$PathToRepo\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.$Service\AzureRM.$Service.psd1"
+
+$HelpPath = "$PathToRepo\src\ResourceManager\$Service\Commands.$Service\help"
+
+# Check to see if there is already a folder containing markdown files for the given service
+if (Test-Path $HelpPath)
+{
+    # Update the markdown files with the changes made in the cmdlets
+    Update-MarkdownHelpModule $HelpPath -AlphabeticParamsOrder
+}
+else
+{
+    Write-Host "Creating folder in $HelpPath to store markdown files"
+    New-Item $HelpPath -ItemType Directory > $null
+
+    # Generate markdown files
+    New-MarkdownHelp -Module AzureRM.$Service -OutputFolder $HelpPath -WithModulePage -AlphabeticParamsOrder
+
+    # Edge case: existing MAML contains parameters WhatIf and Confirm for a cmdlet, but no description
+    # Need to replace the @{Text=} description that will be generated with the auto-generated description from platyPS
+
+    $files = Get-ChildItem -Path $HelpPath
+
+    foreach ($file in $files)
+    {
+        # Ignore the module page
+        if ($file.Name -eq "AzureRM.$Service.md")
+        {
+            continue
+        }
+
+        $content = Get-Content $HelpPath\$file
+        $flag = $False
+
+        $newContent = New-Object string[] $content.Length
+
+        for ($idx = 0; $idx -lt $content.Length; $idx++)
+        {
+            if ($content[$idx] -like "*### -Confirm*" -and $content[$idx + 1] -like "*@{Text=}")
+            {
+                $flag = $True
+                $newContent[$idx] = $content[$idx]
+                $newContent[++$idx] = "Prompts you for confirmation before running the cmdlet."
+                    
+            }
+            elseif ($content[$idx] -like "*### -WhatIf*" -and $content[$idx + 1] -like "*@{Text=}")
+            {
+                $flag = $True
+                $newContent[$idx] = $content[$idx]
+                $newContent[++$idx] = "Shows what would happen if the cmdlet runs. The cmdlet is not run."
+            }
+            else
+            {
+                $newContent[$idx] = $content[$idx]
+            }
+        }
+
+        # Replace the contents of the markdown file if we found an error with the descriptions
+        if ($flag)
+        {
+            $result = $newContent -join "`r`n"
+            $tempFile = Get-Item $HelpPath\$file
+
+            [System.IO.File]::WriteAllText($tempFile.FullName, $result)
+        }
+    }
+}
+
+# Generate the MAML help file from the markdown files
```

Comment:
I liked the idea of one cmdlet to generate your markdown and MAML, no matter your status. Splitting into two cmdlets makes sense. I can change to that model when converting to a module.